scripts/quests/bastok/Out_of_the_Depths.lua

```diff
@@ -0,0 +1,218 @@
+                        else
+                            return quest:progressEvent(308)
+                        end
+                    else
+                        return quest:progressEvent(307)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [307] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:setVar(player, 'Prog', 1)
+                    end
+                end,
+
+                [309] = function(player, csid, option, npc)
+                    print(option)
+                    if option == 1 then
+                        player:addGil(100)
+                        player:messageSpecial(ID.text.GIL_OBTAINED, xi.settings.main.GIL_RATE*100)
+                        player:delKeyItem(xi.ki.DUSTY_TOME)
+                        quest:setVar(player,'Prog', quest:getVar(player, 'Prog') + 1)
+                    elseif option == 2 then
+                        player:addGil(200)
+                        player:messageSpecial(ID.text.GIL_OBTAINED, xi.settings.main.GIL_RATE*200)
+                        player:delKeyItem(xi.ki.POINTED_JUG)
+                        quest:setVar(player,'Prog', quest:getVar(player, 'Prog') + 1)
+                    elseif option == 3 then
+                        player:addGil(300)
+                        player:messageSpecial(ID.text.GIL_OBTAINED, xi.settings.main.GIL_RATE*300)
+                        player:delKeyItem(xi.ki.CRACKED_CLUB)
+                        quest:setVar(player,'Prog', quest:getVar(player, 'Prog') + 1)
+                    elseif option == 4 then
+                        player:addGil(400)
+                        player:messageSpecial(ID.text.GIL_OBTAINED, xi.settings.main.GIL_RATE*400)
+                        player:delKeyItem(xi.ki.PEELING_HAIRPIN)
+                        quest:setVar(player,'Prog', quest:getVar(player, 'Prog') + 1)
+                    elseif option == 5 then
+                        player:delKeyItem(xi.ki.OLD_NAMETAG)
+                        quest:setVar(player,'Prog', 8)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.OLDTON_MOVALPOLOS] =
+        {
+            ['Brakobrik'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(2)
+                    elseif quest:getVar(player, 'Prog') >= 7 then
+                        npc:messageText(npc, 7714)
+                    else
+                        player:messageSpecial(7719, 0, 1669)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if quest:getVar(player, 'Prog') >= 2 and quest:getVar(player, 'Prog') < 7 then
+                        if trade:hasItemQty(1669, 1) and trade:getItemCount() == 1 then
```

Comment:
Seems a bit redundant here, the trade:hasItemQty should be doing the getItemCount as part of it no? Unless I'm not understanding this correctly.

Reply:
to be honest I just copied this from another quest framework. I'll test it without the getItemCount()

Reply:
So what this second check does is prevent players from trading other random items along with the required item. It would then subsequently get gobbled up by the NPC and lost forever. Good check to have! :D